### ISSUE TITLE
Sparkline data goes "out of frame"

### ISSUE THREAD

**Issue by unixb0y** [ratatui-org/ratatui#524]
In essence, as I add data to the vector that serves as an input to a Sparkline, it does *not* automatically make sure that the last item is visible.

It appears to me that as of now, I would have to manually verify that my data structure does not contain too many items, else the latest data won't be visible.

For instance, right now I have a sort of FIFO with 200 elements; after about 180 elements, the box is "full" but it only starts "scrolling" once my actual FIFO is full and the first elements get removed.
Basically, the UI is "left aligned" instead of being "right aligned" (don't confuse this with the *direction* which fills the screen from right to left).

Is this intended behavior or am I missing something?

**Comment by joshka** [comment]
Yes, I think it's intended to be this way. If it wasn't, then people expecting it to not scroll when full would be just as surprised.

App code can fairly easily handle this by taking a slice of the data e.g.:
```rust
let area = ...; // the rect you're drawing into
let data: Vec<u64> = ...; // some data source
let sparkline_data = data[data.len().saturating_sub(area.width)..]; // just the last 180 elements from your example
Sparkline::default().data(sparkline_data).render(area, buf)
```
Alternatively set the direction based on the data length:
```rust
let direction = data.len() < area.width { RenderDirection::LeftToRight } else { RenderDirection::RightToLeft }
```

**Comment by unixb0y** [comment]
I see; I didn't know about this `saturating_sub` method. This works great, thanks a lot!
